fluent-bundle/src/bundle.rs

```diff
@@ -287,19 +287,14 @@ impl<R> FluentBundle<R> {
         &'bundle self,
         pattern: &'bundle ast::Pattern,
         args: Option<&'bundle HashMap<String, FluentValue>>,
-        errors: &mut Vec<FluentError>,
-    ) -> Cow<'bundle, str>
+    ) -> (Cow<'bundle, str>, Vec<FluentError>)
```

Comment:
I remember you said you wanted to wrap the vector of errors in an `Option` to avoid allocation. For that, `Scope` would need to be able to create it lazily. Do you think this is could improve the performance?